Initium-Odp/src/com/universeprojects/miniup/server/commands/CommandForgetAllCombatSites.java

```diff
@@ -0,0 +1,72 @@
+package com.universeprojects.miniup.server.commands;
+
+
+import java.util.List;
+import java.util.Map;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.universeprojects.cacheddatastore.CachedDatastoreService;
+import com.universeprojects.cacheddatastore.CachedEntity;
+import com.universeprojects.miniup.server.GameUtils;
+import com.universeprojects.miniup.server.ODPDBAccess;
+import com.universeprojects.miniup.server.commands.framework.Command;
+import com.universeprojects.miniup.server.commands.framework.UserErrorMessage;
+import com.universeprojects.miniup.server.services.CombatService;
+import com.universeprojects.miniup.server.services.MainPageUpdateService;
+
+/**
+ * Allows the player to attempt to forget all combat sites at their location.
+ * 
+ * @author tacobowl8
+ * 
+ */
+public class CommandForgetAllCombatSites extends Command {
+	
+	public CommandForgetAllCombatSites(ODPDBAccess db, HttpServletRequest request, HttpServletResponse response) {
+		super(db, request, response);
+	}
+	
+	public void run(Map<String, String> parameters) throws UserErrorMessage {
+		ODPDBAccess db = getDB();
+		CachedDatastoreService ds = getDS();
+		CachedEntity character = db.getCurrentCharacter();
+		Key characterLocationKey = (Key)character.getProperty("locationKey");
+		
+		//The location the command is being called from
+		CachedEntity location = db.getLocationById(tryParseId(parameters, "locationId"));
+		
+		if(GameUtils.equals(location.getKey(), characterLocationKey))
+			throw new RuntimeException("Player is not at the location they are forgetting combat sites from.");
+		
+		//We first get all the discoveries for the character and the character's current location.
+		List<CachedEntity> discoveries = db.getDiscoveriesForCharacterAndLocation(character.getKey(), characterLocationKey);
+		ds.beginBulkWriteMode();

```

Comment:
Might as well put the write mode right at the start, in case some of the get calls actually write to the DB.